contracts/custody_beth/src/distribution.rs

```diff
@@ -0,0 +1,157 @@
+use cosmwasm_bignumber::Uint256;
+use cosmwasm_std::{
+    from_binary, log, to_binary, Api, BankMsg, Binary, Coin, CosmosMsg, Env, Extern,
+    HandleResponse, HandleResult, HumanAddr, Querier, QueryRequest, StdError, StdResult, Storage,
+    Uint128, WasmMsg, WasmQuery,
+};
+
+use crate::external::handle::RewardContractHandleMsg;
+use crate::state::{read_config, BETHState, Config};
+
+use cosmwasm_storage::to_length_prefixed;
+use moneymarket::custody::HandleMsg;
+use moneymarket::querier::{deduct_tax, query_all_balances, query_balance};
+use terra_cosmwasm::{create_swap_msg, TerraMsgWrapper};
+
+/// Request withdraw reward operation to
+/// reward contract and execute `distribute_hook`
+/// Executor: overseer
+pub fn distribute_rewards<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+) -> HandleResult<TerraMsgWrapper> {
+    let config: Config = read_config(&deps.storage)?;
+    if config.overseer_contract != deps.api.canonical_address(&env.message.sender)? {
+        return Err(StdError::unauthorized());
+    }
+
+    let reward_contract = deps.api.human_address(&config.reward_contract)?;
+
+    // if there has not been a new reward return.
+    let current_reward_balance = deps
+        .querier
+        .query_balance(reward_contract.clone(), config.stable_denom.as_str())
+        .unwrap_or_default()
+        .amount;
+    let previous_reward_balance = get_previous_balance(deps, reward_contract.clone())?;
+    if current_reward_balance == previous_reward_balance {
```

Comment:
Hmm is this correct flag for no rewards? Only claimer of reward contract is this contract??

if others can claim rewards, this condition can be broken